src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,648 @@
+GWPZ_RBW_HZ = 10.e6
+
+# PPA Area Protection reference bandwidth for the IAP process
+PPA_RBW_HZ = 10.e6
+
+# GWPZ and PPA height (m)
+GWPZ_PPA_HEIGHT = 1.5
+
+# In-band insertion loss
+IN_BAND_INSERTION_LOSS = 0.5
+
+# Define an enumeration class named ProtectedEntityType with members
+# 'GWPZ_AREA', 'PPA_AREA', 'FSS_CO_CHANNEL', 'FSS_BLOCKING', 'ESC'
+
+
+class ProtectedEntityType(Enum):
+  GWPZ_AREA = 1
+  PPA_AREA = 2
+  FSS_CO_CHANNEL = 3
+  FSS_BLOCKING = 4
+  ESC = 5
+
+# Global container to store neighborhood distance type of all the protection 
+_DISTANCE_PER_PROTECTION_TYPE = {
+   ProtectedEntityType.GWPZ_AREA :  (GWPZ_NEIGHBORHOOD_DIST, GWPZ_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.PPA_AREA : ( PPA_NEIGHBORHOOD_DIST,  PPA_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_CO_CHANNEL : ( FSS_CO_CHANNEL_NEIGHBORHOOD_DIST,  FSS_CO_CHANNEL_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_BLOCKING : ( FSS_BLOCKING_NEIGHBORHOOD_DIST,  FSS_BLOCKING_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.ESC: (ESC_NEIGHBORHOOD_DIST_A, ESC_NEIGHBORHOOD_DIST_B)
+    }
+
+# Define CBSD grant, i.e., a tuple with named fields of 'latitude',
+# 'longitude', 'height_agl', 'indoor_deployment', 'antenna_azimuth',
+# 'antenna_gain', 'antenna_beamwidth', 'cbsd_category', 
+# 'max_eirp', 'low_frequency', 'high_frequency', 'is_managed_grant'
+CbsdGrantInformation = namedtuple('CbsdGrantInformation',
+                       ['latitude', 'longitude', 'height_agl', 
+                        'indoor_deployment', 'antenna_azimuth', 'antenna_gain',
+                        'antenna_beamwidth', 'cbsd_category', 
+                        'max_eirp', 'low_frequency', 'high_frequency', 
+                        'is_managed_grant'])
+
+# Define protection constraint, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'low_frequency', 'high_frequency'
+ProtectionConstraint = namedtuple('ProtectionConstraint',
+                                  ['latitude', 'longitude', 'low_frequency',
+                                   'high_frequency', 'entity_type'])
+
+# Define FSS Protection Point, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'height_agl', 'max_gain_dbi', 'pointing_azimuth',
+# 'pointing_elevation' 
+FssProtectionPoint = namedtuple('FssProtectionPoint',
+                                   ['latitude', 'longitude',
+                                    'height_agl', 'max_gain_dbi',
+                                    'pointing_azimuth', 'pointing_elevation'])
+
+
+# Define ESC information, i.e., a tuple with named fields of
+# 'antenna_height', 'antenna_azimuth', 'antenna_gain', 'antenna_pattern_gain'
+EscInformation = namedtuple('EscInformation',
+                                 ['antenna_height', 'antenna_azimuth',
+                                  'antenna_gain', 'antenna_pattern_gain'])
+
+def dbToLinear(x):
+  """This function returns dBm to mW converted value"""
+  return 10**(x / 10)
```

Comment:
Better to put the last 10 as a float, to avoid issues if passing an int to the function:
```
10**(x / 10.)
```

Reply:
Updated   as per suggestion